### ISSUE TITLE
Version issue when building from source archive (version 1.2.3)

### ISSUE THREAD

**Issue by TheChymera** [SimpleITK/SimpleITK#1242]
Seeing as we're having issues with newer lua stability, I have decided to start by packaging version `1.2.3` for Gentoo.

There are a few issues with the build, at this stage 2 Errors (1 for Python and 1 for R), looking at the CMake files, it seems however, that the R issue is related to the missing version warning seen at the start:

```
>>> Unpacking simpleitk-1.2.3.tar.gz to /var/tmp/portage/sci-libs/simpleitk-1.2.3/work
>>> Source unpacked in /var/tmp/portage/sci-libs/simpleitk-1.2.3/work
>>> Preparing source in /var/tmp/portage/sci-libs/simpleitk-1.2.3/work/SimpleITK-1.2.3 ...
 * Working in BUILD_DIR: "/var/tmp/portage/sci-libs/simpleitk-1.2.3/work/simpleitk-1.2.3_build"
 * Hardcoded definition(s) removed in SuperBuild/CMakeLists.txt:
 *    set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR} CACHE PATH "Where all
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/sci-libs/simpleitk-1.2.3/work/SimpleITK-1.2.3 ...
 * Working in BUILD_DIR: "/var/tmp/portage/sci-libs/simpleitk-1.2.3/work/simpleitk-1.2.3_build"
cmake -C /var/tmp/portage/sci-libs/simpleitk-1.2.3/work/simpleitk-1.2.3_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DUSE_SYSTEM_LUA=ON -DUSE_SYSTEM_SWIG=ON -DUSE_SYSTEM_GTEST=ON -DUSE_SYSTEM_ITK=ON -DUSE_SYSTEM_VIRTUALENV=ON -DCMAKE_BUILD_TYPE=Gentoo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/sci-libs/simpleitk-1.2.3/work/simpleitk-1.2.3_build/gentoo_toolchain.cmake  /var/tmp/portage/sci-libs/simpleitk-1.2.3/work/SimpleITK-1.2.3
loading initial cache file /var/tmp/portage/sci-libs/simpleitk-1.2.3/work/simpleitk-1.2.3_build/gentoo_common_config.cmake
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/x86_64-pc-linux-gnu-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/x86_64-pc-linux-gnu-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at CMake/sitkSourceVersion.cmake:41 (message):
  Unable to determine source version!



  Please use the git repository or an official source distribution.

Call Stack (most recent call first):
  Version.cmake:17 (include)
  CMakeLists.txt:23 (include)


-- Building SimpleITK version "1.2.3.dev-gGITDIR-NOTFOUND"
-- Performing Test CXX_HAS_stdcxx11
-- Performing Test CXX_HAS_stdcxx11 - Success
-- Checking if c++11 is required...
-- Checking if c++11 is required... NO
-- Performing Test CXX_HAS-Wall
-- Performing Test CXX_HAS-Wall - Success
-- Performing Test CXX_HAS-Wno-long-double
-- Performing Test CXX_HAS-Wno-long-double - Failed
-- Performing Test CXX_HAS-Wno-long-long
-- Performing Test CXX_HAS-Wno-long-long - Success
-- Performing Test CXX_HAS-Wno-unused-local-typedefs
-- Performing Test CXX_HAS-Wno-unused-local-typedefs - Success
-- Performing Test CXX_HAS-Wno-strict-overflow
-- Performing Test CXX_HAS-Wno-strict-overflow - Success
-- Performing Test CXX_HAS-Wextra
-- Performing Test CXX_HAS-Wextra - Success
-- Performing Test CXX_HAS-Wformat_2
-- Performing Test CXX_HAS-Wformat_2 - Success
-- Performing Test CXX_HAS-Wno-format-nonliteral
-- Performing Test CXX_HAS-Wno-format-nonliteral - Success
-- Performing Test CXX_HAS-Wunused
-- Performing Test CXX_HAS-Wunused - Success
-- Performing Test CXX_HAS-Wpointer-arith
-- Performing Test CXX_HAS-Wpointer-arith - Success
-- Performing Test CXX_HAS-Winvalid-pch
-- Performing Test CXX_HAS-Winvalid-pch - Success
-- Performing Test CXX_HAS-Wcast-align
-- Performing Test CXX_HAS-Wcast-align - Success
-- Performing Test CXX_HAS-Wdisabled-optimization
-- Performing Test CXX_HAS-Wdisabled-optimization - Success
-- Performing Test CXX_HAS-Woverloaded-virtual
-- Performing Test CXX_HAS-Woverloaded-virtual - Success
-- Performing Test CXX_HAS-Wshadow
-- Performing Test CXX_HAS-Wshadow - Success
-- Performing Test CXX_HAS-Wwrite-strings
-- Performing Test CXX_HAS-Wwrite-strings - Success
-- Performing Test CXX_HAS-Wstrict-null-sentinel
-- Performing Test CXX_HAS-Wstrict-null-sentinel - Success
-- Performing Test CXX_HAS-Wno-invalid-offsetof
-- Performing Test CXX_HAS-Wno-invalid-offsetof - Success
-- Performing Test SITK_HAS_CXX11_STATIC_ASSERT
-- Performing Test SITK_HAS_CXX11_STATIC_ASSERT - Success
-- Performing Test SITK_HAS_CXX11_FUNCTIONAL
-- Performing Test SITK_HAS_CXX11_FUNCTIONAL - Success
-- Performing Test SITK_HAS_CXX11_TYPE_TRAITS
-- Performing Test SITK_HAS_CXX11_TYPE_TRAITS - Success
-- Performing Test SITK_HAS_CXX11_UNORDERED_MAP
-- Performing Test SITK_HAS_CXX11_UNORDERED_MAP - Success
-- Performing Test SITK_HAS_CXX11_NULLPTR
-- Performing Test SITK_HAS_CXX11_NULLPTR - Success
-- Performing Test SITK_HAS_CXX11_UNIQUE_PTR
-- Performing Test SITK_HAS_CXX11_UNIQUE_PTR - Success
-- Performing Test SITK_HAS_CXX11_ALIAS_TEMPLATE
-- Performing Test SITK_HAS_CXX11_ALIAS_TEMPLATE - Success
-- Performing Test SITK_HAS_TR1_SUB_INCLUDE
-- Performing Test SITK_HAS_TR1_SUB_INCLUDE - Success
-- Performing Test SITK_HAS_TR1_FUNCTIONAL
-- Performing Test SITK_HAS_TR1_FUNCTIONAL - Success
-- Performing Test SITK_HAS_TR1_TYPE_TRAITS
-- Performing Test SITK_HAS_TR1_TYPE_TRAITS - Success
-- Performing Test SITK_HAS_TR1_UNORDERED_MAP
-- Performing Test SITK_HAS_TR1_UNORDERED_MAP - Success
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
CMake Warning at CMakeLists.txt:319 (message):
  Unable to file ITK's NOTICE File!

   This file will not be included in SimpleITK packaging!


-- Performing Test Weak Link MODULE -> SHARED (gnu_ld_ignore) - Success
CMake Warning (dev) at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:273 (message):
  The package name passed to `find_package_handle_standard_args` (Lua) does
  not match the name of the calling package (LuaInterp).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  CMake/FindLuaInterp.cmake:41 (find_package_handle_standard_args)
  CMake/sitkLanguageOptions.cmake:92 (find_package)
  CMakeLists.txt:340 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Lua: /usr/bin/lua (found version "5.1.5")
-- Found Tclsh: /usr/bin/tclsh (found version "8.6")
CMake Warning (dev) at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:273 (message):
  The package name passed to `find_package_handle_standard_args` (Lua) does
  not match the name of the calling package (LuaInterp).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  CMake/FindLuaInterp.cmake:41 (find_package_handle_standard_args)
  CMake/sitkGenerateFilterSource.cmake:9 (find_package)
  Code/BasicFilters/CMakeLists.txt:4 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Processing json files...
-- Processing json files...done
-- Found SWIG: /usr/bin/swig (found suitable version "4.0.2", minimum required is "3")
-- Found Lua: /usr/lib64/liblua.so;/usr/lib64/libm.so (found version "5.1.5")
-- Found PythonInterp: /var/tmp/portage/sci-libs/simpleitk-1.2.3/temp/python3.7/bin/python (found version "3.7.9")
-- Found PythonVirtualEnv: /usr/bin/virtualenv (found version "virtualenv 20.0.35 from /usr/lib/python3.7/site-packages/virtualenv/__init__.py")
CMake Error at Wrapping/Python/LegacyPackaging.cmake:60 (message):
  In sufficient version of virutalenv for building wheels.  Require
  virtualenv>=13.0.
Call Stack (most recent call first):
  Wrapping/Python/CMakeLists.txt:94 (include)


-- Found TCL: /usr/lib64/libtcl.so
-- Found TCLTK: /usr/lib64/libtcl.so
-- Found TK: /usr/lib64/libtk.so
-- Found R: /usr/lib64/R/include (found version "4.0.2")
CMake Error at Wrapping/R/CMakeLists.txt:64 (math):
  math cannot parse the expression: "9000 + ": syntax error, unexpected $end
  (7).


-- Found GTest: /usr/lib64/libgtest.so
-- Generating tests...
-- Generating tests...Done
-- <<< Gentoo configuration >>>
Build type      Gentoo
Install path    /usr
Compiler flags:
C               -march=native -O3 -pipe
C++             -march=native -O3 -pipe
Linker flags:
Executable      -Wl,-O1 -Wl,--as-needed
Module          -Wl,-O1 -Wl,--as-needed
Shared          -Wl,-O1 -Wl,--as-needed
```

What would be the best way to help it figure out the version? Seeing as this will be a versioned archive, just patching the string into some file would also be an option — if that's the best apparoach, any idea how this should be done?

**Comment by blowekamp** [comment]
>CMake Error at Wrapping/Python/LegacyPackaging.cmake:60 (message):
  In sufficient version of virutalenv for building wheels.  Require
  virtualenv>=13.0.

You can disable using virtualenv with: `SimpleITK_PYTHON_USE_VIRTUALENV:BOOL=OFF`.

> CMake Error at Wrapping/R/CMakeLists.txt:64 (math):
  math cannot parse the expression: "9000 + ": syntax error, unexpected $end
  (7).

I this is coming from the following line:
```
elseif(DEFINED SimpleITK_VERSION_DEV)
  math(EXPR DEV "9000 + ${SimpleITK_VERSION_DEV}")
  set(SimpleITKR_VERSION "${SimpleITKR_VERSION}-${DEV}")
endif()
```
Where `SimpleITK_VERSION_DEV` is defined but an empty string. This is likely an unexpected side effect of building from wrong tar ball. The one Github automatically creates ( and we can't disable from the downloads ) is missing an added version file. Please download the file label "SimpleITK-1.2.3.tar.gz" in the releases section. Any reason not to be using 1.2.4?

**Comment by TheChymera** [comment]
> This is likely an unexpected side effect of building from wrong tar ball.

Oh, indeed. Thank you for this.

> Any reason not to be using 1.2.4?

The two commits which I was linked to as breaking Lua — 77a3d89 and 0dfae3e — were merged 2019-10-22, and 1.2.4 was released on 2019-11-13 . I was assuming `1.2.3` was the last release that worked with older lua.

**Comment by blowekamp** [comment]
> The two commits which I was linked to as breaking Lua — [77a3d89](https://github.com/SimpleITK/SimpleITK/commit/77a3d8943f3c9a4d5b5dcc1452827cb228b7aa2e) and [0dfae3e](https://github.com/SimpleITK/SimpleITK/commit/0dfae3e91dded3eeeb0501b0d00d35f09ed1fe0b) — were merged 2019-10-22, and 1.2.4 was released on 2019-11-13 . I was assuming `1.2.3` was the last release that worked with older lua.

That is not correct. The history is not chronological. Patches were applied to 1.2 while 2.0 was under development.

You can run a command line `git tag --contains 77a3d89` to verify.

**Comment by TheChymera** [comment]
ok, using the proper archive instead of the GitHub autogenerated one fixed this :)
